packages/wonder-blocks-tooltip/components/tooltip-anchor.test.js

```diff
@@ -0,0 +1,560 @@
+        );
+        removeEventListenerSpy.mockClear();
+        const wrapper = mount(nodes);
+
+        // Act
+        wrapper.unmount();
+
+        // Assert
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "focusin",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "focusout",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "mouseenter",
+            expect.any(Function),
+        );
+        expect(removeEventListenerSpy).toHaveBeenCalledWith(
+            "mouseleave",
+            expect.any(Function),
+        );
+    });
+
+    describe("forceAnchorFocusivity is true", () => {
+        test("if not set, sets tabindex on anchor target", (done) => {
+            // Arrange
+            const arrange = (actAndAssert) => {
+                const fakeTooltipPortalMounter = (((
+                    <div>This is the anchor</div>
+                ): any): React.Element<typeof TooltipPortalMounter>);
+                const nodes = (
+                    <div>
+                        <TooltipAnchor
+                            forceAnchorFocusivity={true}
+                            anchorRef={actAndAssert}
+                        >
+                            {(active) => fakeTooltipPortalMounter}
+                        </TooltipAnchor>
+                    </div>
+                );
+                mount(nodes);
+            };
+
+            const actAndAssert = (ref) => {
+                if (!ref) {
+                    return;
+                }
+
+                // Act
+                const result = ref.getAttribute("tabindex");
+
+                // Assert
+                expect(result).toBe("0");
+                done();
+            };
+
+            arrange(actAndAssert);
+        });
+
+        test("if tabindex already set, leaves it as-is", (done) => {
+            // Arrange
+            const arrange = (actAndAssert) => {
+                const fakeTooltipPortalMounter = (((
+                    <div tabIndex="-1">This is the anchor</div>
+                ): any): React.Element<typeof TooltipPortalMounter>);
+                const nodes = (
+                    <div>
+                        <TooltipAnchor
+                            forceAnchorFocusivity={true}
+                            anchorRef={actAndAssert}
+                        >
+                            {(active) => fakeTooltipPortalMounter}
+                        </TooltipAnchor>
+                    </div>
+                );
+                mount(nodes);
+            };
+
+            const actAndAssert = (ref) => {
+                if (!ref) {
+                    return;
+                }
+
+                // Act
+                const result = ref.getAttribute("tabindex");
+
+                // Assert
+                expect(result).toBe("-1");
```

Comment:
I would expect this to be "0" since
> When true, the element wrapped by the anchor will be given tabindex=0

Should the docs or the test be changed?

Reply:
Definitely the docs. We don't want to change the tabindex if it's already been set. The assumption should be that it was set for a good reason and we shouldn't interfere with that.